foreshadow/cleaners/internals/drop.py

```diff
@@ -35,9 +35,9 @@ class DropCleaner(BaseCleaner):
 
     """
 
-    def __init__(self):
+    def __init__(self, column_sharer=None):
```

Comment:
This is a big picture question, but do the individual cleaners need access to the cleaner? From an architectural perspective it might make sense if that as functionality only provided up to the SmartTransformer level to prevent over-reliance on column-sharer. To get around this, classes could define the tags they should set if matched as a class attribute that must be overridden.

Reply:
As the concrete transformers perform the actual transformations, I would see that if they use 'full_df" tag for instance (from v1) then they would need access to the ColumnSharer object. Now, I think we could still move this to the SmartTransformer level by passing them in as kwargs but that might get messy. Let's have a conversation here.

Reply:
Will do.

Reply:
#116  issue